packages/apollo-federation/src/composition/compose.ts

```diff
@@ -432,13 +441,18 @@ export function addFederationMetadataToSchemaNodes({
           providesDirective.arguments &&
           isStringValueNode(providesDirective.arguments[0].value)
         ) {
-          field.federation = {
-            ...field.federation,
+          const fieldFederationMetadata: FederationField = {
+            ...getFederationMetadata(field),
             serviceName,
             provides: parseSelections(
               providesDirective.arguments[0].value.value,
             ),
             belongsToValueType: isValueType,
+          }
+
+          field.extensions = {
+            ...field.extensions,
+            federation: fieldFederationMetadata
```

Comment:
Same concern for creating new `extensions` objects rather than re-cycling the old ones, when present.